proxy/siip_resolve.py

```diff
@@ -16,6 +22,33 @@ def resolve(domain):
         cert_fields['key']
     )
 
-# TODO: resolve the domain name using standard DNS
 def fallback_resolve(domain):
-    return None
\ No newline at end of file
+    # Get Domain's IP
+    # TODO: handle failure
+    ip = socket.gethostbyname(domain)
+
+    # Fetch x509 certificate using OpenSSL
+    # TODO: if this fails, how do we handle it?
+    cert_pem = ssl.get_server_certificate((domain, 443))
```

Comment:
This makes me _very_ nervous. get_server_certificate() returns the certificate without checking it against any certificate authorities. So with a trivial DNS spoof, it's vulnerable to a man-in-the-middle attack. The worst part is that this all happens silently: the browser has no way of knowing if the proxy is using a SIIP certificate or a certificate scraped from the web.

I'm thinking this fallback function needs to move up a function call - i.e. if we can't resolve the domain using standard SIIP, resolve() fails immediately, which relays the error up to the proxy server. The proxy then connects over legacy DNS/TLS, performs the standard handshake and verifies the server's TLS certificate before it ever proceeds with the connection. I've got some of this working locally, but not all of it... lemme push so we can compare code.

This also has implications for your comment on Sam's PR: if we use standard DNS/TLS when the SIIP certificate doesn't exist, we won't need a cache. We just need to do the DNS query (which will already hit a cache somewhere nearby); the server will tell us its certificate anyway.